internal/metrics/metrics.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package metrics
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/open-policy-agent/opa/internal/metrics/prometheus"
+)
+
+// Metrics is the current global metrics provider
+var Metrics Provider
```

Comment:
Can we remove the global/singleton provider in favour of passing it into components that need it? Currently that would be the server and status plugin. The status plugin is instantiated by the discovery package so that would require a small amount of plumbing.

There are several benefits to passing (aka injecting) the provider and the cost is relatively low (i.e., I'd rather plumb the provider into components as needed once in a while than deal with issues in testing, library usage, etc.)

Injecting the provider will also remove the transitive dependency on the prometheus client in this package which we must avoid.